### PR TITLE
Drop EOL django versions and add support for Django 2.1 and Python 3.7 tests

### PR DESCRIPTION
**Pull request by Surgo**
- Drop supports for EOL versions (See: <https://www.djangoproject.com/download/>)
    - Django 1.8
    - Django 1.9
    - Django 1.10
- Add support
    - Django 2.1
    - Python 3.7 (See: <https://docs.djangoproject.com/en/2.1/faq/install/#what-python-version-can-i-use-with-django>)
- Migrate (and cache dependencies) to tox. Env matrix is too complex :(

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,73 +1,60 @@
...
+  - env: TOX_ENV=py34-dj20
+    python: 3.4
+  - env: TOX_ENV=py35-dj20
+    python: 3.5
+  - env: TOX_ENV=py36-dj20
+    python: 3.6
+  - env: TOX_ENV=py37-dj20
+    sudo: true
+    dist: xenial
+    python: 3.7
```
Comment (Surgo) [inline .travis.yml:30]: travis-ci/travis-ci#9815 :(

File: setup.py [hunk 2]
```diff
@@ -1,9 +1,6 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
-#!/usr/bin/env python
```
Comment (Surgo) [inline setup.py:4]: duplicated

File: .travis.yml [hunk 3]
```diff
@@ -1,73 +1,60 @@
...
-# Python versions for matrix
 language: python
-python:
-  - "2.7"
-  - "3.2"
-  - "3.3"
-  - "3.4"
-  - "3.5"
-  - "3.6"
+cache:
```
Comment (Surgo) [inline .travis.yml:5]: Cache dependencies from pypi

**Comment by Surgo** [comment]
Ooops... flake8 issue occured :(

https://travis-ci.org/onysos/django-composite-foreignkey/jobs/413087157

Are you sure you want to fix that in this PR?

**Comment by ornoone** [comment]
hey, thanks for the work ! it look pretty good and clean. I will merge it with a new major release. if you don't mind fixing flake8, you can do it hier, or I will do it myself in days to come.

**Review by ornoone — APPROVED** [review]
LGTM.

**Comment by Surgo** [comment]
@ornoone Thank you for your reviews. I'll fix it :)

File: setup.cfg [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
 [coverage:run]
 omit =
   docs
-  testapp/management
\ No newline at end of file
+  testapp/management
+
+[flake8]
+max-line-length = 119
+exclude =
```
Comment (Surgo) [inline setup.cfg:14]: Ignore automatically generated files

File: setup.cfg [hunk 5]
```diff
@@ -1,9 +1,21 @@
...
 
 [coverage:run]
 omit =
   docs
-  testapp/management
\ No newline at end of file
+  testapp/management
+
+[flake8]
+max-line-length = 119
```
Comment (Surgo) [inline setup.cfg:13]: Copy from tox.ini

**Comment by Surgo** [comment]
Fixed :)

**Review by ornoone — APPROVED** [review]
good job ! thanks for your support

**Comment by ornoone** [comment]
new release 1.1.0 will be pushed by @tomlemeuch  on Monday

**Comment by tohugaby** [comment]
Done
Thank you for your job !

**Pull request merged**